aws_streams/streams_single_region.yaml

```diff
@@ -143,117 +168,115 @@ Resources:
       FirehoseArn: !GetAtt DatadogMetricKinesisFirehose.Arn
       RoleArn: !Ref StreamRoleArn
       OutputFormat: "opentelemetry0.7"
-      IncludeFilters:
-        - !If
-          - HasIncludeNamespace1
-          - Namespace:
-              !Ref FirstNamespace
-          - !Ref 'AWS::NoValue'
-        - !If
-          - HasIncludeNamespace2
-          - Namespace:
-              !Ref SecondNamespace
-          - !Ref 'AWS::NoValue'
-        - !If
-          - HasIncludeNamespace3
-          - Namespace:
-              !Ref ThirdNamespace
-          - !Ref 'AWS::NoValue'
-      ExcludeFilters:
-        - !If
-          - HasExcludeNamespace1
-          - Namespace:
-              !Ref FirstNamespace
-          - !Ref 'AWS::NoValue'
-        - !If
-          - HasExcludeNamespace2
-          - Namespace:
-              !Ref SecondNamespace
-          - !Ref 'AWS::NoValue'
-        - !If
-          - HasExcludeNamespace3
-          - Namespace:
-              !Ref ThirdNamespace
-          - Ref: 'AWS::NoValue'
+      IncludeFilters: !If
+        - IsInclude
+        - - !If
+            - HasIncludeNamespace1
+            - Namespace: !Ref FirstNamespace
+            - !Ref "AWS::NoValue"
```

Comment:
This now only initializes the list of namespaces if `IsInclude` is true, whereas before it would be set to `[]` if there were no namespaces, regardless of the value of `FilterMethod`. 

The addition is the new wrapping condition of `!If IsInclude`, changed from: 
```
      IncludeFilters:
        - !If
          - HasIncludeNamespace1
          - Namespace:
              !Ref FirstNamespace
          - !Ref 'AWS::NoValue'
...
```